gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/CompactionJobConfigurator.java

```diff
@@ -249,6 +250,10 @@ protected boolean configureInputAndOutputPaths(Job job, FileSystemDataset datase
     CompactionPathParser.CompactionParserResult rst = parser.parse(dataset);
     this.mrOutputPath = concatPaths(mrOutputBase, rst.getDatasetName(), rst.getDstSubDir(), rst.getTimeString());
```

Comment:
Can we rewrite this block as a ternary expression instead of having the value of `mrOutputPath` to be bind twice if the condition in the line 253 hold?

Reply:
This is just a short term solution until we have a CompactionDatasetHandler and is easier to remove IMO.